src/mame/drivers/pasha2.cpp

```diff
@@ -130,37 +128,43 @@ class pasha2_state : public driver_device
 	output_finder<3> m_lamps_b;
 
 	// video-related
-	uint8_t m_vbuffer;
+	u8 m_vbuffer = 0;
 
 	// memory
-	std::unique_ptr<uint8_t[]> m_bitmap0[2];
-	std::unique_ptr<uint8_t[]> m_bitmap1[2];
-	void misc_w(offs_t offset, uint16_t data);
-	void palette_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
-	void vbuffer_set_w(uint16_t data);
-	void vbuffer_clear_w(uint16_t data);
-	void bitmap_0_w(offs_t offset, uint8_t data, uint8_t mem_mask = ~0);
-	void bitmap_1_w(offs_t offset, uint8_t data, uint8_t mem_mask = ~0);
-	void pasha2_lamps_w(uint16_t data);
-	uint16_t pasha2_speedup_r(offs_t offset);
-	template<int Chip> void oki_bank_w(offs_t offset, uint16_t data);
-	uint32_t screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
+	bitmap_ind16 m_bg_bitmap[2];
+	bitmap_ind16 m_fg_bitmap[2];
+	void misc_w(offs_t offset, u16 data);
+	void vbuffer_set_w(u16 data);
+	void vbuffer_clear_w(u16 data);
+	void bg_bitmap_w(offs_t offset, u8 data);
+	void fg_bitmap_w(offs_t offset, u8 data);
+
+	void pasha2_lamps_w(u16 data);
+	u16 pasha2_speedup_r(offs_t offset);
+
+	template<int Chip> void oki_bank_w(offs_t offset, u16 data);
+
+	u32 screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
+
+	// devices
 	required_device<cpu_device> m_maincpu;
 	required_device_array<okim6295_device, 2> m_oki;
 	required_device<palette_device> m_palette;
+
+	// address maps
```

Comment:
Is there a reason for leaving private member functions and member variables intermixed like this?  It’s easier to keep track of things if the member variables are kept together.

Reply:
These are leftover from previous code.